packages/x-data-grid-pro/src/DataGridPro/useDataGridProComponent.tsx

```diff
@@ -82,10 +82,8 @@ import {
   rowPinningStateInitializer,
 } from '../hooks/features/rowPinning/useGridRowPinning';
 import { useGridRowPinningPreProcessors } from '../hooks/features/rowPinning/useGridRowPinningPreProcessors';
-import {
-  useGridDataSource,
-  dataSourceStateInitializer,
-} from '../hooks/features/dataSource/useGridDataSource';
+import { useGridDataSourcePro as useGridDataSource } from '../hooks/features/dataSource/useGridDataSourcePro';
```

Comment:
I considered doing that, but for this use-case, it looked to me a bit more verbose since this hook has to be used in the root component(s) and it feels simple enough to just import and use it instead of putting in the context and pulling from there again

I feel configuration is more suitable with hooks or utils that have to be used down the component tree, such as `useGridRowAriaAttributes` used in the `GridRow` component or those that need to be dynamically updated in the lifecycle of a particular Data Grid instance.

I do not have a hard preference on this though. Let me know if you are still unconvinced.

Reply:
let's leave it as is then